src/backend/macos/file_dialog/panel_ffi.rs

```diff
@@ -285,4 +285,46 @@ impl Panel {
 
         Self::new(Retained::into_super(panel), opt.parent.as_ref())
     }
+
+    pub fn build_pick_file_or_folder(opt: &FileDialog, mtm: MainThreadMarker) -> Self {
+        let panel = unsafe { NSOpenPanel::openPanel(mtm) };
+
```

Comment:
If this picker can select files, do filters work for it as well? If so, can we add:
```suggestion

        if !opt.filters.is_empty() {
            panel.add_filters(opt);
        }

```